blues/app-accelerators
# refactor: update nf1 example with new notehub js oauth workflow [DO NOT MERGE]

# Problem Context

The Notehub JS library has been updated to use OAuth bearer tokens for its authentication with the Notehub API, and so our sample code using the library to interact with the Notehub API must be updated accordingly.

## Changes

- Replace HUB_AUTH_TOKEN env var with HUB_CLIENT_ID and HUB_CLIENT_SECRET - these are the two bits of data used to generate an expiring Oauth bearer token via Notehub
- Update the version of the Notehub JS npm library the project is using to take advantage of the new `generateAuthToken()` function and bearer token auth method for all subsequent requests.
- Create a new `generateNotehubAuthToken()` function inside of the `NotehubDataProvider.ts` file that will be called before any other calls to the Notehub API to generate a fresh token for the project and fetch data. Use it in this file and in the `NotehubAttributeStore.ts` file to make updates to the project.
- Remove all now unused references to the HUB_AUTH_TOKEN in the code
- Update `readme.md` file with details on how to get the `client_id` and `client_secret` from a Notehub project for use in the app.

## Screenshot (if applicable)

App is using Notehub JS library to fetch data from Notehub.
![Screenshot 2023-03-23 at 4 32 16 PM](https://user-images.githubusercontent.com/20400845/227348119-618f9ee0-6854-4278-a66b-64ddc630b63b.png)

## Testing

Unit / integration / e2e tests?

All tests pass and code compiles.

Steps to test manually?

1. Grab a `HUB_CLIENT_ID` and `HUB_CLIENT_SECRET` from an Notehub project and add them to the `.env` and `.env.local` project files.
2. Fire up the app and go to http://localhost:4000
3. Route an event or two to the app
4. See the data show up in the UI (the Notehub JS library's taking care of all that)

Any other sorts of testing notes to include?

## Any other related PRs

n/a

## Ticket(s)

n/a
